File: Problems/No.950_Reveal_Cards_In_Increasing_Order/reveal_cards_in_increasing_order.py
class Solution(object):
    def deckRevealedIncreasing(self, deck):
        """
        :type deck: List[int]
        :rtype: List[int]
        """

        # parameters
        ReorderedDeck = [0] * len(deck)
        increasing_deck_idx = range(len(deck))
        
        # start computing
        deck.sort()
        for card in deck:
            ReorderedDeck[increasing_deck_idx[0]] = card
            increasing_deck_idx.pop(0)
            if increasing_deck_idx:
                increasing_deck_idx.append(increasing_deck_idx.pop(0))

        return ReorderedDeck

File: Problems/No.950_Reveal_Cards_In_Increasing_Order/reveal_cards_in_increasing_order.cpp
class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        
        // parameters
        std::vector<int> ReorderedDeck(deck.size(), 0);
        std::queue<int> increasing_deck_idx;

        // start computing
        std::sort(deck.begin(), deck.end());
        for (int i=0; i<deck.size(); i++) increasing_deck_idx.push(i);

        for (auto &card: deck) {

            ReorderedDeck[increasing_deck_idx.front()] = card;
            increasing_deck_idx.pop();
            increasing_deck_idx.push(increasing_deck_idx.front());
            increasing_deck_idx.pop();

        }

        return ReorderedDeck;

    }
};
